samples/file-encryptor/enclave/keys.cpp

```diff
@@ -276,16 +233,45 @@ int ecall_dispatcher::prepare_encryption_header(
     encryption_header_t* header,
     string password)
 {
+    mbedtls_ctr_drbg_context ctr_drbg;
+    mbedtls_entropy_context entropy;
     int ret = 0;
     unsigned char digest[HASH_VALUE_SIZE_IN_BYTES]; // password derived key
     unsigned char
         password_key[ENCRYPTION_KEY_SIZE_IN_BYTES]; // encrypted encryption key
     unsigned char encrypted_key[ENCRYPTION_KEY_SIZE_IN_BYTES];
+    unsigned char salt[SALT_SIZE_IN_BYTES];
+    const char seed[] = "file_encryptor_sample";
+
+    mbedtls_entropy_init(&entropy);
+    mbedtls_ctr_drbg_init(&ctr_drbg);
+
+    // Initialize CTR-DRBG seed
+    ret = mbedtls_ctr_drbg_seed(
+        &ctr_drbg,
+        mbedtls_entropy_func,
+        &entropy,
+        (const unsigned char*)seed,
+        strlen(seed));
+    if (ret != 0)
+    {
+        TRACE_ENCLAVE("mbedtls_ctr_drbg_seed() failed with -0x%04x", -ret);
+        goto exit;
+    }
+
+    // Generate random salt
+    ret = mbedtls_ctr_drbg_random(&ctr_drbg, salt, SALT_SIZE_IN_BYTES);
+    if (ret != 0)
+    {
+        TRACE_ENCLAVE("mbedtls_ctr_drbg_random() failed with -0x%04x", -ret);
+        goto exit;
+    }
+    memcpy(header->salt, salt, SALT_SIZE_IN_BYTES);
```

Comment:
Side note: I just realize that we never check if `header` is null or not. Can you add the check?

Reply:
added null check.